src/service.js

```diff
@@ -443,6 +492,47 @@ function getServiceInternal(holder, context, id, opt_factory,
   return s.obj;
 }
 
+/**
+ * @param {!Object} holder Object holding the service instance.
+ * @param {!Window|!./service/ampdoc-impl.AmpDoc} context Win or AmpDoc.
+ * @param {string} id of the service.
+ * @param {?function(new:Object, ?)=} opt_ctor
+ *     Constructor function to new the service. Called with context.
+ * @param {?function(?)=} opt_factory
+ *     Factory function to create the new service. Called with context.
+ */
+function registerServiceInternal(holder, context, id, opt_ctor, opt_factory) {
+  dev().assert(!opt_factory != !opt_ctor,
+      `Provide a constructor or a factory, but not both for service ${id}`);
+  const services = getServices(holder);
+  let s = services[id];
+
+  if (!s) {
+    s = services[id] = {
+      obj: null,
+      promise: null,
+      resolve: null,
+      build: null,
+    };
+  }
+
+  if (s.build) {
+    // Service already registered
```

Comment:
Added the period. Not sure that warning here is a good idea. We will end up registering a service multiple times anyway since any time a service is requested, we'll need to make sure it's registered first. Registrations after the first will just noop (as they'll bail out on this line)